hazelcast/src/main/java/com/hazelcast/internal/partition/operation/MigrationCommitOperation.java

```diff
@@ -48,6 +50,11 @@ public MigrationCommitOperation(PartitionRuntimeState partitionState) {
 
     @Override
     public void run() {
+        NodeEngine nodeEngine = getNodeEngine();
+        if (!nodeEngine.isRunning()) {
```

Comment:
When a node starts the shut down / termination process, its node state becomes `PASSIVE`. After this moment, `nodeEngine.isRunning()` always returns false. node state becomes `SHUT_DOWN` after the shutdown process is completed.